File: commandfamily-type4/src/test/java/com/taptrack/tcmptappy2/commandfamilies/type4/responses/Type4ErrorResponseTest.java
package com.taptrack.tcmptappy2.commandfamilies.type4.responses;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Type4ErrorResponseTest {
    @Test
    public void testGetCommandCode() throws Exception {
        Type4ErrorResponse response = new Type4ErrorResponse();
        assertEquals(response.getCommandCode(), (byte) 0x7F);
    }

}

File: commandfamily-stmicroM24SR02/src/test/java/com/taptrack/tcmptappy2/commandfamilies/stmicroM24SR02/responses/Type4ErrorResponseTest.kt
package com.taptrack.tcmptappy2.commandfamilies.stmicroM24SR02.responses

import com.taptrack.tcmptappy2.MalformedPayloadException
import org.junit.Assert
import org.junit.Test

class Type4ErrorResponseTest {
    @Test
    @Throws(Exception::class)
    fun testGetCommandCode() {
        val command = Type4ErrorResponse()
        Assert.assertEquals(command.commandCode, 0x02.toByte())
    }

    @Test
    @Throws(Exception::class)
    fun testPayload() {
        val command = Type4ErrorResponse(byteArrayOf(0x00, 0x01))
        Assert.assertArrayEquals(command.payload, byteArrayOf(0x00, 0x01))
    }

    @Test
    @Throws(Exception::class)
    fun testError() {
        Assert.assertThrows(MalformedPayloadException::class.java) {
            val command = Type4ErrorResponse(byteArrayOf(0x00, 0x01, 0x02))
        }
    }
}